### DISCUSSION TITLE
Adding returns to command

### DISCUSSION THREAD

**Discussion by PCanavelli** [fastapi/typer#729] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import random

import typer

from typer.testing import CliRunner

app = typer.Typer()

@app.command()
def main():
    result = random.random()

    return result

def test_app() -> None:
    runner = CliRunner()
    result = runner.invoke(app)

    assert result.return_value == 0
```

### Description

Hi everyone,

I am trying to write some end-to-end tests for a `typer` command that affects the state of an object. 

To so do, I was hoping to set up a unit test using `CliRunner` as per the documentation.

I was expecting the result of the `invoke` method to include the result returned by the invoked command.

However, try as I may, the `return_value` of the `Result` object stays `None`, regardless of whether the command returns something.

Is there a clean way to achieve what is described in the code snippet above?

Cheers

### Operating System

Linux

### Typer Version

0.9.0

### Python Version

3.10.13

**Comment by melsabagh** [comment]
Why not `print` the output then access it in `result.output`?

**Comment by lachaib** [comment, marked as answer]
When working with CLI, the return value is generally discarded. You can pass a return code using `sys.exit` or Typer's equivalent. That way you can assert on the exit code (`result.exit_code`), or `print` (or use `Typer.echo`) and assert on the output `result.output` as suggested by @melsabagh.
Outside of the scope of testing, where would you expect the return of the command to go to?
